src/model/operation/transform.js

```diff
@@ -462,12 +462,20 @@ const ot = {
 			// two separate ranges. Also we have to discard two difference parts.
 			const aCompB = compareArrays( a.sourcePosition.getParentPath(), b.sourcePosition.getParentPath() );
 
-			if ( aCompB == 'prefix' || aCompB == 'extension' ) {
+			if ( aCompB == 'prefix' || aCompB == 'extension' && a.sourcePosition.root == b.sourcePosition.root ) {
 				// Transform `rangeA` by `b` operation and make operation out of it, and that's all.
 				// Note that this is a simplified version of default case, but here we treat the common part (whole `rangeA`)
 				// like a one difference part.
+				//
+				// On a rare occasion, `rangeA` may be collapsed. This means that the move operation moves 0 nodes.
```

Comment:
"On a rare occasion" is my favorite sentence when I am looking a case, where this fix is needed. In 5 years, a guy who will refactor this code will hate you if you will not put there a link to the bug scenario.

Reply:
And note that the guy who will hate you then will be you, most probably :P

Reply:
Phew! For a while I was worried but if that will be me, not much will change :P